transport/src/main/java/io/netty/channel/VoidChannelPromise.java

```diff
@@ -21,7 +21,7 @@
 
 import java.util.concurrent.TimeUnit;
 
-final class VoidChannelPromise extends AbstractFuture<Void> implements ChannelPromise {
+public final class VoidChannelPromise extends AbstractFuture<Void> implements ChannelPromise {
```

Comment:
got it ... not sure if it makes sense to move to an internal package or mark it as unsafe.